src/model/OWL.pl

```diff
@@ -237,30 +237,45 @@
   triple(R,owl:onProperty,P).
 
 is_restriction1(R,value(P,O)) ?+>
-  triple(R,owl:onProperty,P),
-  triple(R,owl:hasValue,O).
+  triple(R,owl:hasValue,O),
```

Comment:
why the reordering?
I think it is more likely that P is ground and that O is var, so it would be better in most cases to leave the ordering like it was.

Reply:
I reordered it in the assumption, that it is more likely that it is called with only R being ground. owl:hasValue (and owl:minQualifiedCardinality etc.) will limit the possible bindings to the exact cases we want, because they are only appearing in the specific type of axioms (e.g. owl:minQualifiedCardinality is only used in qualified minimum cardinality axioms). owl:onProperty is used in several different axiom types (as seen in the other cases).

Reply:
On the higher-level it is convenient to use OWL descriptions in queries, as in:

```
holds(s,p0,value(p1,RangeP1))
```

Then the description is passed through to is_restriction1, I think.
Above is an example of where it might be more efficient to keep the ordering as it was.

Anyway, I think this is subject of query optimization, which we should do during term expansion at some point.

Reply:
Ok, I will return it to the right order then

Reply:
At least it is not obvious to me what would be best, so maybe better leave it like it was for now. Then we can do a more complete investigation of it at some time ;)
As long as it is not causing issues now!